### PR TITLE
Adding the HTTP status health check

### PR DESCRIPTION
**Pull request by nandajavarma**

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -19,26 +19,29 @@
...
+The tool can be used to monitor the HTTP endpoints of a service. It
+check if the service endpoints responds with an HTTP status 200 OK.
+Please note that this tool only supports an in-cluster deployment. The
+tool can be deployed in the following ways:
 
 * As a Kubernetes Job accessing services in the same cluster authorized using RBAC
 
 ## Getting Started
 
+> Please note that this tool doesn't monitor services from outside of
```
Comment (angelbarrera92) [inline README.md:34]: Understood, did you tried with services of type "externalService"? I'm curious about this kind of tests

File: cmd/http-status-check/main.go [hunk 2]
```diff
@@ -89,9 +90,10 @@ func init() {
 	rootCmd.Flags().StringVarP(&serviceName, "service", "s", "", "Name of the service to monitor (required)")
 	rootCmd.Flags().StringVarP(&namespace, "namespace", "n", "default", "Namespace of the service to monitor")
 	rootCmd.Flags().StringVarP(&httpEp, "http-path", "p", "/", "HTTP Path to monitor")
-	rootCmd.PersistentFlags().StringVar(&kubeCfgFile, "KUBECONFIG", "",
-		"kubeconfig file (default is in-cluster connection. Fallback is "+
-			"$HOME/.kube/config")
+	// we do not support out-of-cluster deployment in this
+	// rootCmd.PersistentFlags().StringVar(&kubeCfgFile, "KUBECONFIG", "",
+	// 	"kubeconfig file (default is in-cluster connection. Fallback is "+
+	// 		"$HOME/.kube/config")
```
Comment (angelbarrera92) [inline cmd/http-status-check/main.go:96]: If we don't support this flag, just remove it :)

File: examples/flask-app/cronjob.yaml [hunk 3]
```diff
@@ -0,0 +1,26 @@
+# Copyright (c) 2021 SIGHUP s.r.l All rights reserved.
+# Use of this source code is governed by a BSD-style
+# license that can be found in the LICENSE file.
+---
+apiVersion: batch/v1beta1
```
Comment (angelbarrera92) [inline examples/flask-app/cronjob.yaml:5]: It could be a good idea to use `batch/v1`

File: scripts/e2e/tests.sh [hunk 4]
```diff
@@ -8,38 +8,180 @@ info(){
...
     deploy(){
-        kubectl wait --timeout=180s --for=condition=ready pod -l run=nginx
+        kubectl wait --timeout=180s --for=condition=ready pod -l run=$SVCNAME
     }
     run deploy
     [ "$status" -eq 0 ]
 }
 
-@test "Deploy" {
+@test "Create RBAC for the job " {
```
Comment (angelbarrera92) [inline scripts/e2e/tests.sh:57]: Try to follow the same principle we talked a few days ago.
Run one assertion (when possible) for each test

**Review by angelbarrera92 — CHANGES_REQUESTED** [review]
There are small issues that must be addressed

File: cmd/http-status-check/main.go (hunk 2 above)
Comment (nandajavarma) [reply cmd/http-status-check/main.go]: done! :)

**Pull request merged**